python/lsst/ctrl/bps/wms/panda/panda_service.py

```diff
@@ -154,11 +155,21 @@ def submit(self, workflow):
                 conditions.append(work.is_terminated)
             and_cond = AndCondition(conditions=conditions, true_works=[DAG_final_work])
             idds_client_workflow.add_condition(and_cond)
+        _, idds_server = self.config.search("iddsServer", opt={"default": None})
         c = pandaclient.idds_api.get_api(idds_utils.json_dumps,
-                                         idds_host=self.config.get('idds_server'),
+                                         idds_host=idds_server,
                                          compress=True, manager=True)
-        request_id = c.submit(idds_client_workflow, username=None, use_dataset_name=False)
```

Comment:
"username" can be set from the client. I saw that some requests have set the "username". I am not sure which version it is.

Reply:
@wguanicedew Do you know where the username in the monitor is coming from?  I see my name when I submit even using the main branch which also has username=None.  I just assumed something changed when we switched to the newer iDDS version.  I didn't know if it was coming from my token or from some iDDS/PanDA database or ...  I can set username here, but wasn't sure which username origin was better to use (a user-settable value in BPS or something being done behind the scenes by iDDS or PanDA).

Reply:
I just checked the codes. I found that the username can be left to be empty.
Tadashi had an update in the Panda IAM authentication/relay service to fill the username automatically from the token.
https://github.com/PanDAWMS/panda-server/blob/master/pandaserver/userinterface/UserIF.py#L2594-L2597

Reply:
For now I'll let it get the username from the token.  It will be a simple change to make later if there is a preference or a need to get it from bps.